### ISSUE TITLE
CUDA error - illegal memory access

### ISSUE THREAD

**Issue by kylhuk** [no-fee-ethereum-mining/nsfminer#58]
**Describe the bug**
Every now and then I get to see the following exception in the log which results in stopping the mining completely:
```
 cu-0  Exception thrown in Worker thread: GPU error: CUDA error in func dev::eth::CUDAMiner::search at line 331 an illegal memory access was encountered
```
No other logs available

**To Reproduce**
This bug can't be reproduced manually.

**Expected behavior**
If possible, proper error handling so the miner can continue it's work after the exception happened.

**Environment (please complete the following information):**
- Windows 10 20H2
- RTX 3090
- v1.3.1
- No other options used except the -P flag

**Comment by jean-m-cyr** [comment]
That error message gives me enough to investigate.

**Comment by jean-m-cyr** [comment]
Should be fixed with commit ecb58bc1d9629ed8abad08e1a28357a0629f624e in next release.

**Comment by jean-m-cyr** [comment]
Believed fixed, reopen if not.
